### PR TITLE
swapped simulate() to use simulateOp instead of runOp and added tests

### LINKED ISSUE #2548: Use simulate RPC endpoint instead of run_operation to estimate operations in Taquito
With protocol 17 comes changes to gas_limits for different address curves. 

This introduced a few problems for users or developers that rely on `run_operation` RPC where `gas_exhausted` error will pop up even with a higher gas limit than needed. See this issue for more context: [link](https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/675)

Change the simulation step in Taquito to use the simulate endpoint instead of `run_operation` like it's using right now

### PR DESCRIPTION
**Pull request by dsawali**
closes #2548 

- added `simulateOperation` RPC endpoint
- added unit tests and updated integration tests
- updated `simulate()` in `Provider.ts` to use `simulateOperation()` instead of `runOperation()`

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [x] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [x] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/rpc-nodes.spec.ts [hunk 1]
```diff
@@ -362,6 +362,43 @@ CONFIGS().forEach(
...
+                    gas_limit: '10',
+                    storage_limit: '10',
+                    balance: '0',
+                    script: {
+                      code: tokenBigmapCode,
+                      storage: tokenBigmapStorage,
+                    },
+                  },
+                ],
+                signature:
```
Comment (hui-an-yang) [inline integration-tests/rpc-nodes.spec.ts:389]: signature is optional in simulation request, think it'll be fine without one

File: packages/taquito/src/estimate/rpc-estimate-provider.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export class RPCEstimateProvider extends Provider implements EstimationProvider
       opbytes,
       opOb: { branch, contents },
     } = await this.forge(op);
-    const operation: RPCRunOperationParam = {
+    const operation: RPCSimulateOperationParam = {
       operation: { branch, contents, signature: SIGNATURE_STUB },
```
Comment (hui-an-yang) [inline packages/taquito/src/estimate/rpc-estimate-provider.ts:104]: will we consider removing when simulating signature_stub?

File: packages/taquito-contracts-library/src/rpc-wrapper.ts [hunk 3]
```diff
@@ -225,6 +226,12 @@ export class RpcWrapperContractsLibrary implements RpcClientInterface {
   ): Promise<PreapplyResponse> {
     return this.rpc.runOperation(op, { block });
   }
+  async simulateOperation(
+    op: RPCSimulateOperationParam,
+    { block }: RPCOptions = defaultRPCOptions
+  ): Promise<PreapplyResponse> {
```
Comment (hui-an-yang) [inline packages/taquito-contracts-library/src/rpc-wrapper.ts:232]: surprised to see PreapplyReponse here, but understand if it's returning the same object

Comment (dsawali) [reply packages/taquito-contracts-library/src/rpc-wrapper.ts:232]: PreapplyResponse is basically just `content` so I guess it was just reusing the type. It was the same for run_operation so I just followed suit.

File: packages/taquito/src/estimate/rpc-estimate-provider.ts (hunk 2 above)
Comment (dsawali) [reply packages/taquito/src/estimate/rpc-estimate-provider.ts:104]: good point, removed.

**Review by hui-an-yang — APPROVED** [review]
lgtm 🙌

**Pull request merged**